### PR TITLE
Fix go-next typecheck and add build CI

### PR DESCRIPTION
**Pull request by frzyc**
## Describe your changes

Trying to address the fact that the go-next project has somewhat differing tsconfig, which makes existing libs unbuildable.
- Downgrade to "@types/react": "18.2.68", following advice from `https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/69389` to address the build issue:
image
- In Vite, an image is imported as a `string`, but in nextjs, its imported as `StaticImageData`. This makes type ducking quite weird.
- Add `build go-next` to the CI to ensure further buildability once type issues are addressed.

## Issue or discord link

- 

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ ] I have run `yarn run mini-ci` locally to validate format and lint.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

File: .github/workflows/build-go-next-pr.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    paths-ignore:
+      - 'apps/gi-backend/**'
+      - 'apps/gi-backend-e2e/**'
+      - 'apps/gi-frontend-next-e2e/**'
+      - 'apps/gi-frontend-next-e2e/**'
+      - 'apps/sr-frontend/**'
+      - 'apps/sr-frontend-e2e/**'
+      - 'libs/gi/frontend-gql/**'
+      - 'libs/gi/prisma-schema/**'
+      - 'libs/gi/ui-next/**'
```
Comment (nguyentvan7) [inline .github/workflows/build-go-next-pr.yml:18]: Why ignore ui-next?

File: .github/workflows/build-frontend.yml [hunk 2]
```diff
@@ -48,12 +48,19 @@ jobs:
...
-      - name: Build genshin-optimizer
+      - name: Build frontend
         run: |
           NX_URL_GITHUB_GO_CURRENT_VERSION="https://github.com/${{ github.repository }}/commit/$(git rev-parse HEAD)" \
           NX_DAEMON="false" \
           NX_SHOW_DEV_COMPONENTS="${{inputs.show_dev_components}}" \
-          yarn run nx run ${{inputs.frontend_name}}:build:production
+      - name: Build gi-frontend
+        if: inputs.frontend_name == 'gi-frontend'
+        run: yarn nx run gi-frontend:next:build
```
Comment (nguyentvan7) [inline .github/workflows/build-frontend.yml:58]: Cant you just override build in gi-frontend project.json to call next:build?

File: .github/workflows/build-go-next-pr.yml (hunk 1 above)
Comment (frzyc) [reply .github/workflows/build-go-next-pr.yml:18]: its from the deprecated stack,

File: .github/workflows/build-frontend.yml (hunk 2 above)
Comment (frzyc) [reply .github/workflows/build-frontend.yml:58]: i had trouble getting the `build:production` to work, this is mostly a bandaid anyways

**Review by nguyentvan7 — APPROVED** [review]

**Pull request merged**